gammapy/irf/background.py

```diff
@@ -107,6 +111,23 @@ class Background3D(BackgroundIRF):
     tag = "bkg_3d"
     required_axes = ["energy", "fov_lon", "fov_lat"]
 
+    def __init__(self, axes, data=0, unit="", meta=None, interp_missing_values=False):
+        super().__init__(axes, data, unit, meta)
+        if interp_missing_values:
+            self.interp_missing_values()
+
+    def interp_missing_values(self):
+        """Interpolate missing values in bakcground 3d"""
+        if isinstance(self.axes, list):
+            axis = [a for a in self.axes if a.name == "energy"][0]
+        else:
+            axis = self.axes["energy"]
+        log_energy = np.log(axis.center.value)
```

Comment:
Here we want the 0 to produce NaN not to be clipped to tiny as in the scaling of the `ScaledRegularGridInterpolator()`

Reply:
I guess for the mask creation you don't need to rely on the masked array, you can just do:

```python3

mask = data > 0

interp = ScaledRegularGridInterpolator(values=data[mask], points=(energy[mask]))
interp((energy[~mask])) 

```

Reply:
The `RegularGridInterpolator` won't work in 1d, and using the `masked_invalid` within `.interpolate_invalid_data_3d()` provide a more general solution.

Reply:
The reason why we don't use masked arrays in Gammapy is the performance drop. See e.g.:
- https://github.com/Unidata/netcdf4-python/issues/809
- https://stackoverflow.com/questions/5760668/python-numpy-masked-arrays-are-very-slow

As far as I know this basically common sense and we decided to avoid masked arrays for this reason in general and handle the mask always separately. 

The `ScaledRegularGridInterpolator` just works fine in 1D, see e.g:https://github.com/gammapy/gammapy/blob/master/gammapy/modeling/models/spectral.py#L1315

Reply:
ok done